sumcheck/src/prover/plain.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use crypto::{ElementHasher, RandomCoin};
+use math::FieldElement;
+#[cfg(feature = "concurrent")]
+pub use rayon::prelude::*;
+use smallvec::smallvec;
+
+use super::SumCheckProverError;
+use crate::{
+    comb_func, CompressedUnivariatePolyEvals, FinalOpeningClaim, MultiLinearPoly, RoundProof,
+    SumCheckProof,
+};
+
+/// Sum-check prover for non-linear multivariate polynomial of the simple LogUp-GKR.
+///
+/// More specifically, the following function implements the logic of the sum-check prover as
+/// described in Section 3.2 in [1], that is, given verifier challenges  , the following implements
+/// the sum-check prover for the following two statements
+/// $$
+/// p_{\nu - \kappa}\left(v_{\kappa+1}, \cdots, v_{\nu}\right) = \sum_{w_i}
+///     EQ\left(\left(v_{\kappa+1}, \cdots, v_{\nu}\right), \left(w_{\kappa+1}, \cdots,
+///                              w_{\nu}\right)\right) \cdot
+///      \left( p_{\nu-\kappa+1}\left(1, w_{\kappa+1}, \cdots, w_{\nu}\right)  \cdot
+///              q_{\nu-\kappa+1}\left(0, w_{\kappa+1}, \cdots, w_{\nu}\right) +
+///     p_{\nu-\kappa+1}\left(0, w_{\kappa+1}, \cdots, w_{\nu}\right)  \cdot
+///     q_{\nu-\kappa+1}\left(1, w_{\kappa+1}, \cdots, w_{\nu}\right)\right)
+/// $$
+///
+/// and
+///
+/// $$
+/// q_{\nu -k}\left(v_{\kappa+1}, \cdots, v_{\nu}\right) = \sum_{w_i}EQ\left(\left(v_{\kappa+1},
+///  \cdots, v_{\nu}\right), \left(w_{\kappa+1}, \cdots, w_{\nu }\right)\right) \cdot
+/// \left( q_{\nu-\kappa+1}\left(1, w_{\kappa+1}, \cdots, w_{\nu}\right)  \cdot
+///  q_{\nu-\kappa+1}\left(0, w_{\kappa+1}, \cdots, w_{\nu}\right)\right)
+/// $$
+///
+/// for $k = 1, \cdots, \nu - 1$  
+///
+/// Instead of executing two runs of the sum-check protocol, a batching randomness `r_batch` is
+/// sent by the verifier at the outset in order to batch the two statments.
+///
+/// [1]: https://eprint.iacr.org/2023/1284
+#[allow(clippy::too_many_arguments)]
+pub fn sumcheck_prove_plain<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>>(
+    claim: E,
+    r_batch: E,
+    p0: &mut MultiLinearPoly<E>,
+    p1: &mut MultiLinearPoly<E>,
+    q0: &mut MultiLinearPoly<E>,
+    q1: &mut MultiLinearPoly<E>,
+    eq: &mut MultiLinearPoly<E>,
+    transcript: &mut impl RandomCoin<Hasher = H, BaseField = E::BaseField>,
+) -> Result<SumCheckProof<E>, SumCheckProverError> {
+    let mut round_proofs = vec![];
+
+    let mut claim = claim;
+    let mut challenges = vec![];
+
+    for _ in 0..p0.num_variables() {
+        let len = p0.num_evaluations() / 2;
+
+        #[cfg(not(feature = "concurrent"))]
+        let (eval_point_1, eval_point_2, eval_point_3) =
```

Comment:
```suggestion
        let (round_poly_at_1, round_poly_at_2, round_poly_at_3) =
```
I would make it explicit that the evaluations are ultimately for the round poly. It makes it more clear at this point in the function that we are computing the evaluations of the round polynomial.

Also, I would be helpful to make it explicit in the docstring that the degree of this sumcheck's `g` is 3